File: svws-asd/src/main/java/de/svws_nrw/asd/export/data/KlassenHerkunftStatistikExport.java
package de.svws_nrw.asd.export.data;

import de.svws_nrw.transpiler.TranspilerDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Die Daten zur schulischen Herkunft der (Teil-) Klasse (K82).
 */
@XmlRootElement
@Schema(description = "Die Daten zur schulischen Herkunft der (Teil-) Klasse (K82).")
@TranspilerDTO
public class KlassenHerkunftStatistikExport {

	/** Satzschlüssel: Die Herkunftsschulnummer. */
	@Schema(description = "satzschlüssel: die Herkunftsschulnummer", example = "135443")
	public @NotNull String herkunftsSchulNr = "";

	/** Satzschlüssel: Die Herkunftsschulform. */
	@Schema(description = "satzschlüssel: die Herkunftsschulform", example = "GY")
	public @NotNull String herkunftsschulform = "";

	/** Satzschlüssel: Die Herkunftsart. */
	@Schema(description = "satzschlüssel: die Herkunftsart", example = "11")
	public @NotNull String herkunftsart = "";

	/** Satzschlüssel: Die Grundschulempfehlung. */
	@Schema(description = "satzschlüssel: die Grundschulempfehlung", example = "10XX")
	public @NotNull String kuerzelGrundschuleUebergangsempfehlung = "";

	/** Die Schüler des Herkunftsatzes insgesamt. */
	@Schema(description = "die Schüler des Herkunftsatzes insgesamt.", example = "35")
	public int schuelerInsgesamt = 0;

	/** Die Schüler des Herkunftsatzes weiblich. */
	@Schema(description = "die Schüler des Herkunftsatzes weiblich.", example = "32")
	public int schuelerWeiblich = 0;

	/** Die ausländischen Schüler des Herkunftsatzes zusammen. */
	@Schema(description = "die ausländischen Schüler des Herkunftsatzes zusammen.", example = "29")
	public int schuelerAuslaendischZusammen = 0;

	/** Die ausländischen Schüler des Herkunftsatzes weiblich. */
	@Schema(description = "die ausländischen Schüler des Herkunftsatzes weiblich.", example = "27")
	public int schuelerAuslaendischWeiblich = 0;

	/** Die Anrechnungen des Herkunftssatzes (B-Schulen K86). */
	@Schema(description = "Die Anrechnungen des Herkunftssatzes (B-Schulen K86)")
	public @NotNull KlassenHerkunftAnrechungenStatistikExport klassenHerkunftAnrechungenStatistikExport = new KlassenHerkunftAnrechungenStatistikExport();

	/**
	 * Leerer Standardkonstruktor.
	 */
	public KlassenHerkunftStatistikExport() {
	}

}

File: svws-webclient/core/src/asd/export/data/KlassenHerkunftStatistikExport.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { Class } from '../../../java/lang/Class';
import { KlassenHerkunftAnrechungenStatistikExport } from '../../../asd/export/data/KlassenHerkunftAnrechungenStatistikExport';

export class KlassenHerkunftStatistikExport extends JavaObject {

	/**
	 * Satzschlüssel: Die Herkunftsschulnummer.
	 */
	public herkunftsSchulNr: string = "";

	/**
	 * Satzschlüssel: Die Herkunftsschulform.
	 */
	public herkunftsschulform: string = "";

	/**
	 * Satzschlüssel: Die Herkunftsart.
	 */
	public herkunftsart: string = "";

	/**
	 * Satzschlüssel: Die Grundschulempfehlung.
	 */
	public kuerzelGrundschuleUebergangsempfehlung: string = "";

	/**
	 * Die Schüler des Herkunftsatzes insgesamt.
	 */
	public schuelerInsgesamt: number = 0;

	/**
	 * Die Schüler des Herkunftsatzes weiblich.
	 */
	public schuelerWeiblich: number = 0;

	/**
	 * Die ausländischen Schüler des Herkunftsatzes zusammen.
	 */
	public schuelerAuslaendischZusammen: number = 0;

	/**
	 * Die ausländischen Schüler des Herkunftsatzes weiblich.
	 */
	public schuelerAuslaendischWeiblich: number = 0;

	/**
	 * Die Anrechnungen des Herkunftssatzes (B-Schulen K86).
	 */
	public klassenHerkunftAnrechungenStatistikExport: KlassenHerkunftAnrechungenStatistikExport = new KlassenHerkunftAnrechungenStatistikExport();


	/**
	 * Leerer Standardkonstruktor.
	 */
	public constructor() {
		super();
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.export.data.KlassenHerkunftStatistikExport';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.export.data.KlassenHerkunftStatistikExport'].includes(name);
	}

	public static readonly class = new Class<KlassenHerkunftStatistikExport>('de.svws_nrw.asd.export.data.KlassenHerkunftStatistikExport');

	public static transpilerFromJSON(json: string): KlassenHerkunftStatistikExport {
		const obj = JSON.parse(json) as Partial<KlassenHerkunftStatistikExport>;
		const result = new KlassenHerkunftStatistikExport();
		if (obj.herkunftsSchulNr === undefined)
			throw new Error('invalid json format, missing attribute herkunftsSchulNr');
		result.herkunftsSchulNr = obj.herkunftsSchulNr;
		if (obj.herkunftsschulform === undefined)
			throw new Error('invalid json format, missing attribute herkunftsschulform');
		result.herkunftsschulform = obj.herkunftsschulform;
		if (obj.herkunftsart === undefined)
			throw new Error('invalid json format, missing attribute herkunftsart');
		result.herkunftsart = obj.herkunftsart;
		if (obj.kuerzelGrundschuleUebergangsempfehlung === undefined)
			throw new Error('invalid json format, missing attribute kuerzelGrundschuleUebergangsempfehlung');
		result.kuerzelGrundschuleUebergangsempfehlung = obj.kuerzelGrundschuleUebergangsempfehlung;
		if (obj.schuelerInsgesamt === undefined)
			throw new Error('invalid json format, missing attribute schuelerInsgesamt');
		result.schuelerInsgesamt = obj.schuelerInsgesamt;
		if (obj.schuelerWeiblich === undefined)
			throw new Error('invalid json format, missing attribute schuelerWeiblich');
		result.schuelerWeiblich = obj.schuelerWeiblich;
		if (obj.schuelerAuslaendischZusammen === undefined)
			throw new Error('invalid json format, missing attribute schuelerAuslaendischZusammen');
		result.schuelerAuslaendischZusammen = obj.schuelerAuslaendischZusammen;
		if (obj.schuelerAuslaendischWeiblich === undefined)
			throw new Error('invalid json format, missing attribute schuelerAuslaendischWeiblich');
		result.schuelerAuslaendischWeiblich = obj.schuelerAuslaendischWeiblich;
		if (obj.klassenHerkunftAnrechungenStatistikExport === undefined)
			throw new Error('invalid json format, missing attribute klassenHerkunftAnrechungenStatistikExport');
		result.klassenHerkunftAnrechungenStatistikExport = KlassenHerkunftAnrechungenStatistikExport.transpilerFromJSON(JSON.stringify(obj.klassenHerkunftAnrechungenStatistikExport));
		return result;
	}

	public static transpilerToJSON(obj: KlassenHerkunftStatistikExport): string {
		let result = '{';
		result += '"herkunftsSchulNr" : ' + JSON.stringify(obj.herkunftsSchulNr) + ',';
		result += '"herkunftsschulform" : ' + JSON.stringify(obj.herkunftsschulform) + ',';
		result += '"herkunftsart" : ' + JSON.stringify(obj.herkunftsart) + ',';
		result += '"kuerzelGrundschuleUebergangsempfehlung" : ' + JSON.stringify(obj.kuerzelGrundschuleUebergangsempfehlung) + ',';
		result += '"schuelerInsgesamt" : ' + obj.schuelerInsgesamt.toString() + ',';
		result += '"schuelerWeiblich" : ' + obj.schuelerWeiblich.toString() + ',';
		result += '"schuelerAuslaendischZusammen" : ' + obj.schuelerAuslaendischZusammen.toString() + ',';
		result += '"schuelerAuslaendischWeiblich" : ' + obj.schuelerAuslaendischWeiblich.toString() + ',';
		result += '"klassenHerkunftAnrechungenStatistikExport" : ' + KlassenHerkunftAnrechungenStatistikExport.transpilerToJSON(obj.klassenHerkunftAnrechungenStatistikExport) + ',';
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

	public static transpilerToJSONPatch(obj: Partial<KlassenHerkunftStatistikExport>): string {
		let result = '{';
		if (obj.herkunftsSchulNr !== undefined) {
			result += '"herkunftsSchulNr" : ' + JSON.stringify(obj.herkunftsSchulNr) + ',';
		}
		if (obj.herkunftsschulform !== undefined) {
			result += '"herkunftsschulform" : ' + JSON.stringify(obj.herkunftsschulform) + ',';
		}
		if (obj.herkunftsart !== undefined) {
			result += '"herkunftsart" : ' + JSON.stringify(obj.herkunftsart) + ',';
		}
		if (obj.kuerzelGrundschuleUebergangsempfehlung !== undefined) {
			result += '"kuerzelGrundschuleUebergangsempfehlung" : ' + JSON.stringify(obj.kuerzelGrundschuleUebergangsempfehlung) + ',';
		}
		if (obj.schuelerInsgesamt !== undefined) {
			result += '"schuelerInsgesamt" : ' + obj.schuelerInsgesamt.toString() + ',';
		}
		if (obj.schuelerWeiblich !== undefined) {
			result += '"schuelerWeiblich" : ' + obj.schuelerWeiblich.toString() + ',';
		}
		if (obj.schuelerAuslaendischZusammen !== undefined) {
			result += '"schuelerAuslaendischZusammen" : ' + obj.schuelerAuslaendischZusammen.toString() + ',';
		}
		if (obj.schuelerAuslaendischWeiblich !== undefined) {
			result += '"schuelerAuslaendischWeiblich" : ' + obj.schuelerAuslaendischWeiblich.toString() + ',';
		}
		if (obj.klassenHerkunftAnrechungenStatistikExport !== undefined) {
			result += '"klassenHerkunftAnrechungenStatistikExport" : ' + KlassenHerkunftAnrechungenStatistikExport.transpilerToJSON(obj.klassenHerkunftAnrechungenStatistikExport) + ',';
		}
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

}

export function cast_de_svws_nrw_asd_export_data_KlassenHerkunftStatistikExport(obj: unknown): KlassenHerkunftStatistikExport {
	return obj as KlassenHerkunftStatistikExport;
}
